lib/router.js

```diff
@@ -421,8 +421,9 @@ Router.prototype._handlePaths = function(rootNode, spec, scope) {
 
     // Handle paths
     // Sequence the build process with `.each` to avoid race conditions
-    // while building the tree.
-    return P.each(Object.keys(paths), function(pathPattern) {
+    // while building the tree. Also sort(), so that fixed path segments sort
+    // before patterns (by virtue of `[a-zA-Z/] < '{'`).
+    return P.each(Object.keys(paths).sort(), function(pathPattern) {
```

Comment:
I think the grouping by shared path prefix aligns with related functionality, so is good. The main downside to alphabetical sorting I see is that we don't have control over sorting by importance, like sorting data-parsoid lower than html. However, with multiple schemas that is hard to control, anyway.

In any case, we can decouple the documentation sort order from processing order by sorting a copy of the array instead. This leaves the spec itself unchanged, so the current sort order would be preserved.